bin/collator/src/parachain/service.rs

```diff
@@ -405,25 +405,23 @@ where
         RelayChainError::ServiceError(polkadot_service::Error::Sub(x)) => x,
         s => format!("{}", s).into(),
     })?;
-    let block_announce_validator = BlockAnnounceValidator::new(relay_chain_interface.clone(), id);
 
     let force_authoring = parachain_config.force_authoring;
     let is_authority = parachain_config.role.is_authority();
     let prometheus_registry = parachain_config.prometheus_registry().cloned();
     let transaction_pool = params.transaction_pool.clone();
     let import_queue_service = params.import_queue.service();
     let (network, system_rpc_tx, tx_handler_controller, start_network) =
-        sc_service::build_network(sc_service::BuildNetworkParams {
-            config: &parachain_config,
+        cumulus_client_service::build_network(BuildNetworkParams {
+            parachain_config: &parachain_config,
+            para_id: id,
             client: client.clone(),
             transaction_pool: transaction_pool.clone(),
             spawn_handle: task_manager.spawn_handle(),
             import_queue: params.import_queue,
-            block_announce_validator_builder: Some(Box::new(|_| {
-                Box::new(block_announce_validator)
-            })),
-            warp_sync_params: None,
-        })?;
+            relay_chain_interface: relay_chain_interface.clone(),
+        })
+        .await?;
```

Comment:
Just interested, how come we can get rid of `block_announce_validator_builder`? Is it not required or deprecated?